src/python/pants/backend/project_info/filter_targets.py

```diff
@@ -136,3 +139,22 @@ def warn_deprecated_target_type(tgt_type: type[Target]) -> None:
         entity=f"using `--filter-target-type={tgt_type.deprecated_alias}`",
         hint=f"Use `--filter-target-type={tgt_type.alias}` instead.",
     )
+
+
+class FilterGoal(Goal):
+    subsystem_cls = FilterSubsystem
+
+
+@goal_rule
+def filter_targets(
+    addresses: Addresses, filter_subsystem: FilterSubsystem, console: Console
+) -> FilterGoal:
+    # `SpecsFilter` will have already filtered for us.
```

Comment:
A deprecation would be good here (targeted a year out perhaps?), as the new syntax is definitely more powerful, and users might not find it otherwise.